SampleAPI.UnitTest/ControllerTests/EmployeeControllerTest.cs

```diff
@@ -0,0 +1,288 @@
+using Moq;
+using SampleAPI.Constant;
+using SampleAPI.Controllers;
+using SampleAPI.Models;
+using SampleAPI.Repository;
+using Xunit;
+
+namespace SampleAPI.UnitTest.ControllerTests
+{
+    public class EmployeeControllerTest
+    {
+        public readonly Fixture _fixture;
+        private readonly Mock<ILogger<EmployeeController>> _loggerMock;
+        private readonly Mock<IEmployeeRepository> _employeeRepositoryMock;
+
+        public EmployeeControllerTest()
+        {
+            _fixture = new Fixture();
+            _loggerMock = new Mock<ILogger<EmployeeController>>();
+            _employeeRepositoryMock = new Mock<IEmployeeRepository>();
+        }
+
+        private EmployeeController CreateController()
+        {
+            return new EmployeeController(_loggerMock.Object, _employeeRepositoryMock.Object);
+        }
+
+        #region GetAllEmployee 
+
+        [Fact]
+        public async Task ShouldReturn_500InternalServerErrorMessage_WhenEmployeeReposistoryThrowException_WhileGetEmployeeIsCalled()
+        {
+            //Arrange
+            var search = _fixture.Create<string>();
+            var role = _fixture.Create<string>();
+            var getEmployeeParameters = _fixture.Create<GetEmployeeParameters>();
+            _employeeRepositoryMock.Setup(x => x.GetAllAsync(getEmployeeParameters)).ThrowsAsync(_fixture.Create<Exception>());
+            var controller = CreateController();
+
+            //Act
+            var actual = await controller.GetEmployee(getEmployeeParameters) as ObjectResult;
+
+            //Assert
+            Assert.NotNull(actual);
+            Assert.Equal(StatusCodes.Status500InternalServerError, actual.StatusCode);
+            _employeeRepositoryMock.Verify(m => m.GetAllAsync(getEmployeeParameters), Times.Once);
+            _employeeRepositoryMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldReturn_AllEmployee_WhenGetEmployeeIsCalled()
+        {
+            //Arrange
+            var search = _fixture.Create<string>();
+            var role = _fixture.Create<string>();
+            var getEmployeeParameters = _fixture.Build<GetEmployeeParameters>().With(x=>x.PageSize, 0).Create();
+            var expected = _fixture.Create<IEnumerable<EmployeeModel>>().AsQueryable();
+            _employeeRepositoryMock.Setup(x => x.GetAllAsync(getEmployeeParameters)).ReturnsAsync(expected);
+            var controller = CreateController();
+
+            //Act
+            var actual = await controller.GetEmployee(getEmployeeParameters) as ObjectResult;
+
+            //Assert
+            Assert.NotNull(actual);
+            Assert.Equal(StatusCodes.Status200OK, actual.StatusCode);
+            Assert.Equal(expected, (actual.Value as GetResponseModel<EmployeeModel>).Data);
+            _employeeRepositoryMock.Verify(m => m.GetAllAsync(getEmployeeParameters), Times.Once);
+            _employeeRepositoryMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
```

Comment:
Fact attribute has name field that can be used for giving a string description of the tests like
[Fact(Name="")]

And so the method name can be simple like GivenPaginationParameters_ReturnsExpectedResult